modules/ding_react/ding_react.module

```diff
@@ -384,65 +491,75 @@ function ding_react_user_migrate(stdClass $account) {
-        'Authorization' => "Bearer ${token}",
-      ]
-    ];
-
-    $material_list_url = variable_get('ding_react_material_list_url');
-    $response = drupal_http_request($material_list_url . '/migrate/' . $legacy_uid,
-      $migrate_request_options);
-    if (!empty($response->error)) {
-      watchdog(
-        'ding_react',
-        'Unable to migrate user data in material list: (%code) %message %data',
-        [
-          '%code' => $response->code,
-          '%message' => $response->error,
-          '%data' => $response->data
-        ],
-        WATCHDOG_ERROR
-      );
-      return;
-    }
+  if (empty($legacy_uid)) {
+    ding_react_log('User not migrated. No registered OpenList uid for user.');
+    return FALSE;
+  }
+  if (empty($token)) {
+    ding_react_log('User not migrated. No token available');
+    return FALSE;
+  }
+  if (strpos($legacy_uid, DING_REACT_MIGRATED_UID_PREFIX) === 0) {
+    ding_react_log('User migration skipped. User has already been migrated previously.');
+    // We return true here because nothing failed even if the user was not
+    // actually migrated.
+    return TRUE;
+  }
 
-    $follow_searches_url = variable_get('ding_react_follow_searches_url');
-    $response = drupal_http_request($follow_searches_url . '/migrate/' . $legacy_uid,
-      $migrate_request_options);
-    if (!empty($response->error)) {
-      watchdog(
-        'ding_react',
-        'Unable to migrate user data in Follow Searches: (%code) %message %data',
-        [
-          '%code' => $response->code,
-          '%message' => $response->error,
-          '%data' => $response->data
-        ],
-        WATCHDOG_ERROR
-      );
-      return;
-    }
+  $migrate_request_options = [
+    'method' => 'PUT',
+    // Use a relatively low timeout. We do not what login to stall if services
+    // are slow to respond. Then migration will have to wait.
+    'timeout' => variable_get('ding_react_migrate_timeout', 5),
+    'headers' => [
+      'Authorization' => "Bearer ${token}",
+    ]
+  ];
 
-    // Add a prefix to show the user has been migrated. This way we do not
-    // throw away data yet but avoid migrating users multiple times.
-    $account->openlist_uid = $migrated_prefix . $legacy_uid;
-    user_save($account);
+  $material_list_url = variable_get('ding_react_material_list_url');
+  $response = drupal_http_request($material_list_url . '/migrate/' . $legacy_uid,
+    $migrate_request_options);
+  if (!empty($response->error)) {
+    ding_react_log(
+      'Unable to migrate user data in material list: (%code) %message %data',
+      [
+        '%code' => $response->code,
+        '%message' => $response->error,
+        '%data' => $response->data
+      ],
+      WATCHDOG_ERROR
+    );
+    return FALSE;
+  }
 
-    watchdog(
-      'ding_react',
-      'Migrated user data for material list og follow searches',
-      [],
-      WATCHDOG_NOTICE
+  $follow_searches_url = variable_get('ding_react_follow_searches_url');
```

Comment:
Why not use `ding_react_follow_searches_url()` to get the URL which has fallback URL?